src/timedelta_isoformat/__init__.py

```diff
@@ -85,33 +101,47 @@ def _from_designators(duration: str) -> Components:
         in order of largest-to-smallest unit from left-to-right (with the exception of
         week measurements, which must be the only measurement in the string if present).
         """
-        date_tokens = iter(("Y", "years", "M", "months", "D", "days"))
-        time_tokens = iter(("H", "hours", "M", "minutes", "S", "seconds"))
-        week_tokens = iter(("W", "weeks"))
-
-        tokens, value = date_tokens, ""
+        date_context = ParsingContext(
+            {
+                "Y": "years",
+                "M": "months",
+                "D": "days",
+            }
+        )
+        time_context = ParsingContext(
+            {
+                "H": "hours",
+                "M": "minutes",
+                "S": "seconds",
+            }
+        )
+        week_context = ParsingContext(
+            {"W": "weeks"}
+        )
+
+        context, value = date_context, ""
         for char in duration:
             if char in _FORMAT:
                 value += char
                 continue
 
-            if char == "T" and tokens is not time_tokens:
-                tokens, value = time_tokens, ""
+            if char == "T" and context is not time_context:
```

Comment:
I like the renaming of token-related variables to include the word `context` :+1: